Tasks/DockerInstallerV0/task.json

```diff
@@ -0,0 +1,59 @@
+{
+    "id": "8e038650-f7bf-11e8-8a6c-8fff434b4eff",
+    "name": "DockerInstaller",
+    "friendlyName": "Docker CLI installer",
+    "description": "Install Docker CLI on agent machine.",
+    "helpMarkDown": "[More Information](https://go.microsoft.com/fwlink/?linkid=851275)",
+    "category": "Tool",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 0,
+        "Minor": 1,
+        "Patch": 0
+    },
+    "demands": [],
+    "satisfies": [
+        "Docker"
+    ],
+    "minimumAgentVersion": "2.115.0",
+    "inputs": [
+        {
+            "name": "dockerVersion",
+            "label": "Docker Version",
+            "type": "string",
+            "required": true,
+            "helpMarkDown": "Specify the version of Docker CLI to install",
+            "defaultValue": "17.09.0-ce"
```

Comment:
Is there any REST API to fetch versions?
You could populate the versions in a dropdown if there is

Reply:
AFAIK, there is no REST API or a file to parse for latest version.

Reply:
You could parse the GitHub releases stream and pull the latest non-beta version. 

https://github.com/docker/docker-ce/releases

I've seen the trick below to pull the latest release file/archive using `jq`, but you could probably work around needing `jq` with a clever combination of `curl` or `wget` filters.

```
 # If you have jq installed this will automatically find the latest release binary for your architecture and download it
    curl --silent "https://api.github.com/repos/docker/compose/releases/latest" | jq --arg PLATFORM_ARCH "$(echo `uname -s`-`uname -m`)" -r '.assets[] | select(.name | endswith($PLATFORM_ARCH)).browser_download_url' | xargs sudo curl -L -o /usr/local/bin/docker-compose --url
```